### ISSUE TITLE
kustomize build: could not find expected ':'

### ISSUE THREAD

**Issue by zhouhaibing089** [kubernetes-sigs/kustomize#3626]
`kustomize build` complains invalid YAML even though the YAML is well formatted(for e.g, `yamllint` reports no error ).

### kustomization.yaml

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- resources.yaml
```

### resources.yaml

```yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: tekton-pipelines-leader-election
  labels:
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-pipelines
rules:
- # We uses leases for leaderelection
  apiGroups: ["coordination.k8s.io"]
  resources: ["leases"]
  verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
```

**Expected output**

```console
$ kustomize build
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-pipelines
  name: tekton-pipelines-leader-election
rules:
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - list
  - create
  - update
  - delete
  - patch
  - watch
```

**Actual output**

```console
$ kustomize build
Error: map[string]interface {}{"apiVersion":"rbac.authorization.k8s.io/v1", "kind":"ClusterRole", "metadata":map[string]interface {}{"labels":map[string]interface {}{"app.kubernetes.io/instance":"default", "app.kubernetes.io/part-of":"tekton-pipelines"}, "name":"tekton-pipelines-leader-election"}, "rules":[]interface {}{map[string]interface {}{"apiGroups":[]interface {}{"coordination.k8s.io"}, "resources":[]interface {}{"leases"}, "verbs":[]interface {}{"get", "list", "create", "update", "delete", "patch", "watch"}}}}: yaml: line 11: could not find expected ':'
```

**Kustomize version**

```console
$ kustomize version
{Version:kustomize/v4.0.1 GitCommit:516ff1fa56040adc0173ff6ece66350eb4ed78a9 BuildDate:2021-02-14T05:00:02+00:00 GoOs:darwin GoArch:amd64}
```

**Platform**

macOS

**Additional context**

`kustomize build` works when I change the `resources.yaml` to be:

```yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: tekton-pipelines-leader-election
  labels:
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-pipelines
rules:
# We uses leases for leaderelection
- apiGroups: ["coordination.k8s.io"]
  resources: ["leases"]
  verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
```

See the difference of wherer `# We uses leases for leaderelection` is positioned.

By the way, kustomize v2.0.3 works fine and we see this issue when we upgrade to the latest version.

**Comment by Shell32-Natsu** [comment]
Similar with #3614. This should be fixed after #3271.

**Comment by zhouhaibing089** [comment]
v4.4.0 fixed this issue - https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.4.0
